common/interpreter/gipsy_scheme/packages/oops-package.scm

```diff
@@ -63,11 +63,11 @@
 
    (define (_check-class sym c)
      (if (not (class? c))
-         (error "argument is not a class")))
+         (catch-throw::throw "argument is not a class")))
 
```

Comment:
Just for my understanding, 1. this catch-throw::throw is more like an "error reporting" rather than catching an exception and then throwing. 2. From the definition of catch-throw::throw, I see that apart from a  _pop-handler, I sense that throw x is same as error x, not sure what the _pop-handler is doing though.